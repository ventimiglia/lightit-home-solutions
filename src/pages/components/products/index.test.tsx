import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";
import Products from "./";

const queryClient = new QueryClient();

describe("Products", () => {
  it("displays sidebar categories", () => {
    const { container, getByText } = render(
      <QueryClientProvider client={queryClient}>
        <Products />
      </QueryClientProvider>
    );

    const openingCategory = getByText("aberturas");
    const equipmentCategory = getByText("equipamiento");
    const finishesCategory = getByText("terminaciones");

    expect(openingCategory).toBeInTheDocument();
    expect(equipmentCategory).toBeInTheDocument();
    expect(finishesCategory).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
