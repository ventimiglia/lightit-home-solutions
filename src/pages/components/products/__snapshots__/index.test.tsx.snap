// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products displays sidebar categories 1`] = `
<div>
  <main
    class="flex flex-1"
  >
    <section
      class="flex flex-col justify-center items-center w-28 h-full bg-white"
    >
      <div
        class="flex flex-col items-center justify-center w-full relative h-24 cursor-pointer z-10"
      >
        <img
          class="transition duration-600 ease-in-out transform hover:scale-110"
          src="opening.png"
          width="55"
        />
        <span
          class="text-sm text-primary capitalize"
        >
          aberturas
        </span>
      </div>
      <div
        class="flex flex-col items-center justify-center w-full relative h-24 cursor-pointer z-10"
      >
        <img
          class="transition duration-600 ease-in-out transform hover:scale-110"
          src="equipment.png"
          width="55"
        />
        <span
          class="text-sm text-primary capitalize"
        >
          equipamiento
        </span>
      </div>
      <div
        class="flex flex-col items-center justify-center w-full relative h-24 cursor-pointer z-10"
      >
        <img
          class="transition duration-600 ease-in-out transform hover:scale-110"
          src="finishes.png"
          width="55"
        />
        <span
          class="text-sm text-primary capitalize"
        >
          terminaciones
        </span>
      </div>
    </section>
    <aside
      class="flex flex-col justify-between flex-1 items-end p-4 relative"
    >
      <div
        class="flex gap-4 absolute bottom-4"
      >
        <div
          class="flex flex-col flex-wrap justify-between bg-white rounded-md p-1"
        >
          <button
            class="transition duration-300 ease-in-out transform hover:scale-125"
          >
            <img
              alt="plus"
              src="plus.png"
              width="22"
            />
          </button>
          <hr />
          <button
            class="transition duration-300 ease-in-out transform hover:scale-125"
          >
            <img
              alt="less"
              src="less.png"
              width="22"
            />
          </button>
        </div>
        <div
          class="w-16 h-16 relative bg-white border-4 border-white rounded-md"
        >
          <button
            class="w-5 h-5 absolute top-0 right-1/2 translate-x-1/2 transition duration-300 ease-in-out transform hover:scale-125"
            name="up"
          >
            <img
              alt="up"
              src="arrow.png"
            />
          </button>
          <button
            class="w-5 h-5 absolute right-0 top-1/2 -translate-y-1/2 transition duration-300 ease-in-out transform hover:scale-125"
            name="right"
          >
            <img
              alt="right"
              class="rotate-90"
              src="arrow.png"
            />
          </button>
          <button
            class="w-5 h-5 absolute left-0 top-1/2 -translate-y-1/2 transition duration-300 ease-in-out transform hover:scale-125"
            name="left"
          >
            <img
              alt="left"
              class="-rotate-90"
              src="arrow.png"
            />
          </button>
          <button
            class="w-5 h-5 absolute bottom-0 left-1/2 -translate-x-1/2 transition duration-300 ease-in-out transform hover:scale-125"
            name="down"
          >
            <img
              alt="down"
              class="rotate-180"
              src="arrow.png"
            />
          </button>
        </div>
      </div>
    </aside>
  </main>
</div>
`;
